src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewAddColumnDialog.cs

```diff
@@ -0,0 +1,996 @@
+                        DataGridViewDesigner.ShowErrorDialog(uiService, errorString, _liveDataGridView);
+                    }
+
+                    return true;
+            }
+        }
+
+        return base.ProcessDialogKey(keyData);
+    }
+
+    internal void Start(int insertAtPosition, bool persistChangesToDesigner)
+    {
+        _insertAtPosition = insertAtPosition;
+        _persistChangesToDesigner = persistChangesToDesigner;
+
+        if (_persistChangesToDesigner)
+        {
+            _initialDataGridViewColumnsCount = _liveDataGridView!.Columns.Count;
+        }
+        else
+        {
+            _initialDataGridViewColumnsCount = -1;
+        }
+    }
+
+    private void SetDefaultDataGridViewColumnType(Type type)
+    {
+        TypeConverter imageTypeConverter = TypeDescriptor.GetConverter(typeof(Image));
+
+        if (type == typeof(bool) || type == typeof(CheckState))
+        {
+            // get the data grid view check box column type
+            _columnTypesCombo!.SelectedIndex = TypeToSelectedIndex(typeof(DataGridViewCheckBoxColumn));
+        }
+        else if (typeof(Image).IsAssignableFrom(type) || imageTypeConverter.CanConvertFrom(type))
+        {
+            // get the data grid view image column type
+            _columnTypesCombo!.SelectedIndex = TypeToSelectedIndex(typeof(DataGridViewImageColumn));
+        }
+        else
+        {
+            // get the data grid view text box column type
+            _columnTypesCombo!.SelectedIndex = TypeToSelectedIndex(typeof(DataGridViewTextBoxColumn));
+        }
+    }
+
+    private int TypeToSelectedIndex(Type type)
+    {
+        for (int i = 0; i < _columnTypesCombo!.Items.Count; i++)
+        {
+            if (type == (_columnTypesCombo!.Items![i] as ComboBoxItem)!.ColumnType)
+            {
+                return i;
+            }
+        }
+
+        Debug.Assert(false, "we should have found a type by now");
+
+        return -1;
+    }
+
+    public static bool ValidName(string name,
+        DataGridViewColumnCollection columns,
+        IContainer? container,
+        INameCreationService? nameCreationService,
+        DataGridViewColumnCollection liveColumns,
+        bool allowDuplicateNameInLiveColumnCollection)
+    {
+        if (columns.Contains(name))
+        {
+            return false;
+        }
+
+        if (container is not null && container.Components[name] is not null)
+        {
+            if (!allowDuplicateNameInLiveColumnCollection || liveColumns is null || !liveColumns.Contains(name))
+            {
+                return false;
+            }
+        }
+
+        if (nameCreationService is not null && !nameCreationService.IsValidName(name))
+        {
+            if (!allowDuplicateNameInLiveColumnCollection || liveColumns is null || !liveColumns.Contains(name))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

Comment:
This method and the method below are almost exactly the same. The only difference seems to be that the one below also has an `out string errorString` in its signature. 
It is hard to tell from looking on github, but could you see if it is necessary to keep both around? Could we just remove this method and update any calls to this to use the method below? 
If there are a lot of callers to this one then alternatively let's update this method to look like the following:

```suggestion
    public static bool ValidName(string name,
        DataGridViewColumnCollection columns,
        IContainer? container,
        INameCreationService? nameCreationService,
        DataGridViewColumnCollection liveColumns,
        bool allowDuplicateNameInLiveColumnCollection)
            => ValidName(name, columns, container, nameCreationService, liveColumns, allowDuplicateNameInLiveColumnCollection, out _);
```